atomic_reactor/util.py

```diff
@@ -1611,3 +1611,12 @@ def get_platform_config(platform, build_annotations):
     cm_key = cm_key_tmp[cmlen:]
 
     return cm_key, cm_frag_key
+
+
+def chain_get(d, keys, default=None):
+    for k in keys:
+        try:
+            d = d[k]
```

Comment:
@ben-alkov; This function is shamelessly stolen from osbs-client (with minor changes). I think you have the right idea about what it does, but I'll try to give an example just to be sure:

Consider the following yaml file:
```yaml
spec:
  relatedImages:
    - name: foo
      image: registry/foo:1
```
Using `jq`, you would query for the image pullspec like this (after converting from yaml to json)
```bash
$ cat file.json | jq '.spec.relatedImages[0].image'
"registry/foo:1"
```
With this function, you can achieve the same thing for a python dict:
```python
>>> chain_get(data, ["spec", "relatedImages", 0, "image"])
"registry/foo:1"

# basically the same as
data["spec"]["relatedImages"][0]["image"]
# but returns None if any of the lookups would fail
```

> Also, single letter variable names :rage:

`d` as a variable name for "this is a dictionary with futher unspecified content" is pretty common, but it's true that the meaning may change inside the loop (sometimes, it may be a `list`, at the end, it will usually be a `str`, `int`, `float` or `bool`). I could rename to something that better reflects that, but best I can think of is `obj` for object

Reply:
@chmeliik; Very clear, thanks!